src/applications/disability-benefits/all-claims/components/ArrayField.jsx

```diff
@@ -0,0 +1,428 @@
+  // in the array. This shouldn’t be necessary, but there’s a fix in rjsf
+  // that has not been released yet
+  componentDidMount() {
+    const { schema, formData = [], registry } = this.props;
+    if (schema.minItems > 0 && formData.length === 0) {
+      this.props.onChange(
+        Array(schema.minItems).fill(
+          getDefaultFormState(
+            schema.additionalItems,
+            undefined,
+            registry.definitions,
+          ),
+        ),
+      );
+    }
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return !deepEquals(this.props, nextProps) || nextState !== this.state;
+  }
+
+  onItemChange(indexToChange, value) {
+    const newItems = _.set(indexToChange, value, this.props.formData || []);
+    this.props.onChange(newItems);
+  }
+
+  getItemSchema(index) {
+    const schema = this.props.schema;
+    if (schema.items.length > index) {
+      return schema.items[index];
+    }
+
+    return schema.additionalItems;
+  }
+
+  scrollToTop() {
+    setTimeout(() => {
+      scroller.scrollTo(
+        `topOfTable_${this.props.idSchema.$id}`,
+        window.Forms.scroll || {
+          duration: 500,
+          delay: 0,
+          smooth: true,
+          offset: -60,
+        },
+      );
+    }, 100);
+  }
+
+  scrollToRow(id) {
+    setTimeout(() => {
+      scroller.scrollTo(
+        `table_${id}`,
+        window.Forms.scroll || {
+          duration: 500,
+          delay: 0,
+          smooth: true,
+          offset: 0,
+        },
+      );
+    }, 100);
+  }
+
+  /*
+   * Clicking edit on an item that’s not last and so is in view mode
+   */
+  handleEdit(index, status = true) {
+    this.setState(_.set(['editing', index], status, this.state), () => {
+      this.scrollToRow(`${this.props.idSchema.$id}_${index}`);
+    });
+  }
+
+  /*
+   * Clicking Update on an item that’s not last and is in edit mode
+   */
+  handleUpdate(index) {
+    if (errorSchemaIsValid(this.props.errorSchema[index])) {
+      this.setState(_.set(['editing', index], false, this.state), () => {
+        this.scrollToTop();
+      });
+    } else {
+      // Set all the fields for this item as touched, so we show errors
+      const touched = setArrayRecordTouched(this.props.idSchema.$id, index);
+      this.props.formContext.setTouched(touched, () => {
+        scrollToFirstError();
+      });
+    }
+  }
+
+  /*
+   * Clicking Save
+   */
+  handleSave() {
+    const lastIndex = this.props.formData.length - 1;
+    if (errorSchemaIsValid(this.props.errorSchema[lastIndex])) {
+      // When we add another, we want to change the editing state of the currently
+      // last item, but not ones above it
+      const newEditing = this.state.editing.map(
+        (val, index) => (index + 1 === this.state.editing.length ? false : val),
+      );
+      const newState = _.assign(this.state, {
+        editing: newEditing,
+      });
+
+      this.setState(newState);
```

Comment:
This looks safe (and is how the other handlers do it) so probably not a real issue but I think [it'd be safer](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous) to pass a function instead since new state is derived from old state. (`newEditing = this.state.editing.map(...`). Maybe something like:

`this.setState((state) => ({ editing: state.editing.map((val, index) => (index + 1 === this.state.editing.length ? false : val)) });`

or more legibly maybe something like:
```
this.setState((state) => {
  const newEditing = state.editing.map(
    (val, index) => (index + 1 === this.state.editing.length ? false : val),
  );
  
  return { editing: newEditing };
}
```